ReactAndroid/src/main/java/com/facebook/react/views/text/ReactBaseTextShadowNode.java

```diff
@@ -86,15 +87,23 @@ private static void buildSpannedFromShadowNode(
       ReactBaseTextShadowNode textShadowNode,
       SpannableStringBuilder sb,
       List<SetSpanOperation> ops,
+      TextAttributes parentTextAttributes,
       int start) {
 
+    TextAttributes textAttributes;
+    if (parentTextAttributes != null) {
```

Comment:
Probably we can simplify logic if `parentTextAttributes` always be non-null. So we can implement `TextAttributes::defaultTextAttributes()` (with default values, obviously) and pass into the function initially.

I am not sure about this though. I am also curious how I did this in Fabric.

Reply:
If I had a `defaultTextAttributes()`, I could get rid of the `parentTextAttributes == null` checks.

Some tricky things to watch out for:
  - What if the data type doesn't have a good value to represent "not set" (everything is `float` now and `float` has `Float.NaN`)?
  - We have to ensure that the root text shadow node always adds `fontSize` to the span even if it matches the default value (`ViewDefaults.FONT_SIZE_SP`).